functions/utils/useCache.ts

```diff
@@ -0,0 +1,15 @@
+export async function getCache(request: string, name: string) {
+  const cache = await caches.open(name)
+  const response = await cache.match(request)
+  if (response) {
+    const data = JSON.parse(await response.text())
+    return data
+  }
+  return undefined
+}
+
+export async function putCache(response: Response, request: string, name: string) {
```

Comment:
Does the cached object _need_ to be a `Response`, because you're not actually dealing with `Response` objects.

Reply:
I believe so

<img width="689" alt="Screenshot 2023-08-03 at 10 51 51 AM" src="https://github.com/Uniswap/interface/assets/35351983/982b4c43-805d-4228-a510-4d6b7fe09640">